core/src/main/scala/org/bitcoins/core/psbt/PSBTHelper.scala

```diff
@@ -0,0 +1,225 @@
+package org.bitcoins.core.psbt
+
+import org.bitcoins.core.crypto.{ECDigitalSignature, ECPublicKey, ExtPublicKey}
+import org.bitcoins.core.hd.BIP32Path
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.CompactSizeUInt
+import org.bitcoins.core.protocol.script.{
+  ScriptPubKey,
+  ScriptSignature,
+  WitnessScriptPubKey
+}
+import org.bitcoins.core.protocol.transaction.{Transaction, WitnessTransaction}
+import org.bitcoins.core.psbt.GlobalPSBTRecord._
+import org.bitcoins.core.psbt.InputPSBTRecord._
+import org.bitcoins.core.psbt.PSBTGlobalKey._
+import org.bitcoins.core.psbt.PSBTInputKey._
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.core.serializers.script.RawScriptWitnessParser
+import org.bitcoins.core.util.BitcoinScriptUtil
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+object PSBTHelper {
+
+  private def parseKeyAndValue(
+      bytes: ByteVector): (Long, ByteVector, ByteVector) = {
+    val keySize = CompactSizeUInt.parse(bytes)
+    val key = bytes.drop(keySize.size).take(keySize.toLong)
+    val valueSize =
+      CompactSizeUInt.parse(bytes.drop(keySize.size + keySize.toLong))
+    val value = bytes
+      .drop(keySize.size + keySize.toLong + valueSize.size)
+      .take(valueSize.toLong)
+    val totalSize = keySize.size + keySize.toLong + valueSize.size + valueSize
+      .toLong
+    (totalSize, key, value)
+  }
+
+  @tailrec
+  final def parseGlobalMap(
+      remainingBytes: ByteVector,
+      accum: Seq[GlobalPSBTRecord],
+      tx: Transaction): (ByteVector, Seq[GlobalPSBTRecord], Transaction) = {
+    if (remainingBytes.head == 0x00.byteValue) // terminator
```

Comment:
this is unsafe, as we don't know if `remaingBytes.nonEmpty`

Reply:
while scala does allow you to not use `{}` to wrap if expressions, I think it is best practice to use this syntax 

```scala
if (...whatever...) { 
  //if stuff in here 
} else { 
  //else stuff in here
}
```

Reply:
Also we should definitely have some `val` some where to represent the terminator rather than just having `0x00` hard coded everywhere.